website/blog/2018-05-23-1.13.0.md

```diff
@@ -0,0 +1,534 @@
+
+When using the API, `prettier.format(text, { options })` will throw an error if either `parser` or `filepath` are specified; it will also throw an error if no `parser` is specified, `filepath` is specified but no parser can be inferred from the path extension.
+
+#### prettier.getFileInfo() method and --file-info CLI option ([#4341](https://github.com/prettier/prettier/pull/4341) by [@kachkaev](https://github.com/kachkaev))
+
+When invoking Prettier from the API, there was no way to find out if a file should be ignored or not (for `.prettierignore`) so editor integration plugins would have to implement this logic. Also, there was no way to find out if a file is supported by Prettier, so users would need to call Prettier anyway. Now in 1.13 there's a way to check if a parser will be inferred (and which one) and if the file is ignored or not before actually calling `format`.
+
+You can read more about it in the [docs](https://prettier.io/docs/en/api.html#prettiergetfileinfofilepath-options).
+
+### JavaScript
+
+#### Inline AngularJS tests that use `inject` ([#4495](https://github.com/prettier/prettier/pull/4495) by [@thorn0](https://github.com/thorn0))
+
+Prettier will now keep AngularJS tests with dependency injection in a single line, like it does for our other tests.
+
+<!-- prettier-ignore -->
+```js
+// Input:
+it("has calculated the answer correctly", inject(function(DeepThought) {
+  expect(DeepThought.answer).toBe(42);
+}));
+
+// Output with Prettier 1.12.1:
+it(
+  "has calculated the answer correctly",
+  inject(function(DeepThought) {
+    expect(DeepThought.answer).toBe(42);
+  })
+);
+
+// Output with Prettier 1.13:
+it("has calculated the answer correctly", inject(function(DeepThought) {
+  expect(DeepThought.answer).toBe(42);
+}));
+```
+
+#### Nicer line wrapping for d3 ([#4285](https://github.com/prettier/prettier/pull/4285) by [@1wheel](https://github.com/1wheel), [#4505](https://github.com/prettier/prettier/pull/4505) by [@duailibe](https://github.com/duailibe))
+
+In Prettier 1.12 chains that started with `d3` and other short names were broken in too many lines. Prettier 1.13 will now detect if the first member of the chain is a short identifier and prevent breaking.
+
+<!-- prettier-ignore -->
+```js
+// Input
+d3.select('body')
+  .append('circle')
+  .at({ width: 30, fill: '#f0f' })
+  .st({ fontWeight: 600 })
+
+// Output with Prettier 1.12.1:
+d3
+  .select("body")
+  .append("circle")
+  .at({ width: 30, fill: "#f0f" })
+  .st({ fontWeight: 600 });
+
+// Output with Prettier 1.13:
+d3.select("body")
+  .append("circle")
+  .at({ width: 30, fill: "#f0f" })
+  .st({ fontWeight: 600 });
+```
+
+#### Support jest-each template string ([#4423](https://github.com/prettier/prettier/pull/4423) by [@ikatyang](https://github.com/ikatyang))
+
+With the recent addition of support for [data driven tests](https://github.com/facebook/jest/pull/6102) in Jest (to be released), Prettier will format the matrix!
```

Comment:
Hooooopefully this week, we'll see what we can to at the Jest summit now.

It's available in the best (`jest@beta` though), if you wanna link that

Reply:
And again, much ❤️ for implementing support for the table